From 17569fd41bd61fd892ae1af1e15e3fbe0045726a Mon Sep 17 00:00:00 2001
From: Aleksander Morgado <aleksander@aleksander.es>
Date: Mon, 11 Jan 2021 14:13:28 +0100
Subject: [PATCH] telit: plug GPS data port leak

The _get_port_gps() returns a full reference, use _peek_port_gps()
instead.

See https://gitlab.freedesktop.org/mobile-broadband/ModemManager/-/issues/302

(cherry picked from commit 102de14c15b1c278988de376d03262bcb078211d)
---
 plugins/telit/mm-broadband-modem-telit.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/plugins/telit/mm-broadband-modem-telit.c b/plugins/telit/mm-broadband-modem-telit.c
index 8fc869c1..d4bcdf13 100644
--- a/plugins/telit/mm-broadband-modem-telit.c
+++ b/plugins/telit/mm-broadband-modem-telit.c
@@ -359,7 +359,7 @@ gpsp_test_ready (MMIfaceModemLocation *self,
     if (error) {
         mm_obj_dbg (self, "GPS controller not supported: %s", error->message);
         g_clear_error (&error);
-    } else if (mm_base_modem_get_port_gps (MM_BASE_MODEM (self)))
+    } else if (mm_base_modem_peek_port_gps (MM_BASE_MODEM (self)))
         sources |= (MM_MODEM_LOCATION_SOURCE_GPS_NMEA |
                     MM_MODEM_LOCATION_SOURCE_GPS_RAW  |
                     MM_MODEM_LOCATION_SOURCE_GPS_UNMANAGED);
-- 
2.25.1

